docs/core/04-contracts-overview.md

```diff
@@ -0,0 +1,117 @@
+| `Plugin`                         | Intended for non-upgradeable plugin implementations deployed via the `new` keyword.                                                                              | implements `IPlugin`                 |
+| `PluginCloneable`                | Intended for non-upgradeable plugin implementations deployed via the minimal clones pattern.                                                                     | implements `IPlugin`                 |
+| `PluginUUPSUpgradeable`          | Intended for upgradeable plugin implementations deployed via the UUPS pattern.                                                                                   | implements `IPlugin`                 |
+| `DaoAuthorizableBase`            | Provides a `DAO` variable, setter, and meta-transaction compatible modifier for non-upgradeable contracts to authorize function calls through an associated DAO. | used by `DaoAuthorizable`            |
+| `DaoAuthorizable`                | Adds a constructor to `DaoAuthorizableBase`.                                                                                                                     | used by `Plugin`                     |
+| `DaoAuthorizableCloneable`       | Adds an internal initializer to `DaoAuthorizableBase`.                                                                                                           | used by `PluginCloneable`            |
+| `DaoAuthorizableBaseUpgradeable` | Provides a `DAO` variable, setter, and meta-transaction compatible modifier for upgradeable contracts to authorize function calls through an associated DAO.     | used by `DaoAuthorizableUpgradeable` |
+| `DaoAuthorizableUpgradeable`     | Adds an internal initializer method and storage gap to `DaoAuthorizableBaseUpgradeable`.                                                                         | used by `PluginUUPSUpgradeable`      |
+
+### Framework Infrastructure
+
+#### DAO Creation
+
+| Contract      | Description                                                | Relationship    |
+| :------------ | :--------------------------------------------------------- | :-------------- |
+| `DAOFactory`  | A template to setup plugins for a DAO using our framework. | creates `DAO`   |
+| `DAORegistry` | Registers DAOs                                             | registers `DAO` |
+
+#### Plugin Management
+
+| Contract             | Description                                                         | Relationship                                         |
+| :------------------- | :------------------------------------------------------------------ | :--------------------------------------------------- |
+| `PluginSetup`        | A template to setup plugins for a DAO using our framework.          | `Plugin`, `PluginCloneable`, `PluginUUPSUpgradeable` |
```

Comment:
```suggestion
| `PluginSetup`        | A template so that plugin developers can bootstrap plugins and helpers for a DAO using our framework.          | `Plugin`, `PluginCloneable`, `PluginUUPSUpgradeable` |
```